ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +227,289 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+
+        # use this user in any other entity so you can login with it
+        self.login_page.login(username=user_random_name, password='1')
+
+        # make sure that I can login with the user that I created it
+        time.sleep(15)
+        self.assertIn('dashboard', self.login_page.base_selenium.get_url())
+
+        self.header_page.click_on_header_button()
+        self.base_selenium.click(element='header:user_management_button')
+        self.base_page.sleep_small()
+
+        self.base_selenium.LOGGER.info(
+            'search to make sure after I login the user found in the active tabele '
+                .format(user_random_name))
+        self.header_page.search(value=user_random_name)
+
+        self.header_page.select_all_records()
+        # navigate to the archived table to delete it
+        self.header_page.archive_entity(menu_element='user_management:right_menu',
+                                        archive_element='user_management:archive')
+        self.header_page.get_archived_entities(menu_element='user_management:right_menu',
+                                               archived_element='user_management:archived')
+
+        self.header_page.select_all_records()
+        self.base_selenium.LOGGER.info('Press on the right menu')
+        self.base_selenium.click(element='user_management:right_menu')
+        self.base_selenium.LOGGER.info('Press on the delete button')
+        self.base_selenium.click(element='user_management:delete')
+        self.header_page.confirm_popup()
+
+        self.base_selenium.LOGGER.info(
+            'message will appear this user related to some data & cant delete it')
+        self.header_page.confirm_popup()
+
+        self.base_selenium.LOGGER.info(
+            'search to make sure this user found in the active table '.format(user_random_name))
+        result = self.header_page.search(value=user_random_name)
+        self.assertTrue(result, user_data)
+
+    def test013_filter_by_user_name(self):
+        """
+        User management Approach: I can filter by user name successfully
+        LIMS-6002
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+        random_user_name = self.generate_random_string()
+        random_user_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_user_email,
+                                         user_role='', user_password='1', user_confirm_password='1')
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        user_filter = self.header_page.filter_user_by(filter_element='user_management:filter_name',
+                                                      filter_text=random_user_name)
+        result_user = self.header_page.result_table()[0]
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
```

Comment:
what if the test case failed?! 
this should be in tearDown and u need to check if the runnign test case is test013_filter_by_user_name then reset the filter

Reply:
@dianaeldemerdash ??